CRM/Contribute/BAO/ContributionRecur.php

```diff
@@ -996,10 +996,11 @@ public static function updateOnTemplateUpdated(CRM_Contribute_DAO_Contribution $
       ->first();
 
     if ($contribution->total_amount === NULL) {
+      // What does this do? Why is it here?
```

Comment:
I think it's probably a wise precaution - since this is being called from a `pre` hook. I practice the contribution create function probably fully loads the contribution cos it's just that kinda function - but it's not an explicit expectation of the contract so it's possible either `$contribution->total_amount` or `$contribution->currency` could be blank at this point